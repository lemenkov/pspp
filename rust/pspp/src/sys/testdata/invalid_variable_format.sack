# File header.
"$FL2"; s60 "$(#) SPSS DATA FILE PSPP synthetic test file";
2; 4; 1; 0; -1; 100.0; "01 Jan 11"; "20:53:52"; s64 ""; i8 0 *3;

# Numeric variable, invalid format types.
# No warning is issued for type 0 because it has been observed in real
# system files.
2; 0; 0; 0; >>0xff0800; 0<<; s8 "NUM1";

# Numeric variable, string formats.
2; 0; 0; 0; >>0x010800<<; >>0x021000<<; s8 "VAR1";

# String variable, numeric formats.
2; 4; 0; 0; >>0x050800<<; >>0x110a01<<; s8 "STR1";

# String variable, wrong width formats.
2; 4; 0; 0; >>0x010800<<; >>0x020400<<; s8 "STR2";

# Character encoding record.
7; 20; 1; 12; "windows-1252";

# End of dictionary.
999; 0;
