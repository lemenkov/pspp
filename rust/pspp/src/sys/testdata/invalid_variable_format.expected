Substituting F8.2 for invalid print format on variable NUM1.  Unknown format type 255.

Substituting F8.2 for invalid write format on variable NUM1.  Unknown format type 0.

Substituting F8.2 for invalid print format on variable VAR1.  Numeric variable is not compatible with string format A.

Substituting F8.2 for invalid write format on variable VAR1.  Numeric variable is not compatible with string format AHEX.

Substituting A4 for invalid print format on variable STR1.  String variable is not compatible with numeric format F.

Substituting A4 for invalid write format on variable STR1.  String variable is not compatible with numeric format E.

Substituting A4 for invalid print format on variable STR2.  String variable with width 4 is not compatible with format A8.  Use format A4 instead.

Substituting A4 for invalid write format on variable STR2.  String variable with width 4 is not compatible with format AHEX4.  Use format AHEX8 instead.

╭──────────────────────┬────────────────────────╮
│       Created        │    01-JAN-2011 20:53:52│
├──────────────────────┼────────────────────────┤
│Writer Product        │PSPP synthetic test file│
├──────────────────────┼────────────────────────┤
│       Compression    │SAV                     │
│       Number of Cases│Unknown                 │
╰──────────────────────┴────────────────────────╯

╭─────────┬─╮
│Variables│4│
╰─────────┴─╯

                                               Variables
╭────┬────────┬─────┬─────────────────┬─────┬─────┬─────────┬────────────┬────────────┬──────────────╮
│    │Position│Label│Measurement Level│ Role│Width│Alignment│Print Format│Write Format│Missing Values│
├────┼────────┼─────┼─────────────────┼─────┼─────┼─────────┼────────────┼────────────┼──────────────┤
│num1│       1│     │                 │Input│    8│Right    │F8.2        │F8.2        │              │
│var1│       2│     │                 │Input│    8│Right    │F8.2        │F8.2        │              │
│str1│       3│     │Nominal          │Input│    4│Left     │A4          │A4          │              │
│str2│       4│     │Nominal          │Input│    4│Left     │A4          │A4          │              │
╰────┴────────┴─────┴─────────────────┴─────┴─────┴─────────┴────────────┴────────────┴──────────────╯
